homeassistant/bootstrap.py

```diff
@@ -402,7 +402,13 @@ def log_exception(ex, domain, config):
                    .format(ex.path[-1], domain, domain,
                            '->'.join('%s' % m for m in ex.path))
     else:
-        message += humanize_error(config, ex)
+        try:
+            message += humanize_error(config, ex)
+        except TypeError:
+            # Shorten ex.path - can be caused by a vol.All(ensure_list, <fail>)
```

Comment:
Validating `{Key: None}` with `vol.Schema({Key: vol.All(ensure_list, [str])})`

The None will not be a string so the `path=['Key', 0]` -- the first element in the array of key
So `None[0]` raises TypeError. By popping 1 valie off the end of path, we get a error on Key instead of a unhandled TypeError exception